symfit/core/fit_results.py

```diff
@@ -171,13 +172,18 @@ def _array_safe_dict_eq(one_dict, other_dict):
         """
         for key in one_dict:
             try:
-                if key == 'objective' or key == 'minimizer':
+                if key == 'minimizer':
                     assert one_dict[key].__class__ == other_dict[key].__class__
                 elif key == 'minimizer_output':
                     # Ignore this, because it can contain unexpected terms and
                     # if all the derived attributes are correct I see no reason
                     # why this term shouldn't be at least close enough.
                     pass
+                elif key == 'constraints':
+                    assert len(one_dict['constraints']) == len(other_dict['constraints'])
+                    for constr1, constr2 in zip(one_dict['constraints'],
+                                                other_dict['constraints']):
+                        assert constr1 == constr2
```

Comment:
How is this different from `assert one_dict['constraints'] == other_dict['constraints']`? And should the order of the constraints matter?

Reply:
Hmmm, it no longer is. That's because I first hadn't defined `__eq__` for objectives. Good catch.

Reply:
I'll update it to a set compare to ignore ordering as well.